### DISCUSSION TITLE
Pricing and Licensing Plans

### DISCUSSION THREAD

**Discussion by SamTolmay** [lowdefy/lowdefy#1668] [Announcement]
## Bootstrapping Lowdefy

To date, we have bootstrapped the development of Lowdefy by running a software consulting business. We have used Lowdefy in the products we create for customers, and invested back into the framework where possible. This helped us develop solutions to real-world problems and test our codebase in complex production environments, however, delivering our consulting service can be demanding and often absorbed all our limited resources.

We have built Lowdefy as an Open-Source project with the goal of following an open-core model, creating additional value-added services to further fund the development of the project. As we started the development of some of these value-added services, such as hosting or analytics, we realised it was distracting us, but also that some of these services are often much better served by other market leaders. We’ve come to realise that our time would be much better spent on improving the framework. 

## Monetising The Value We Create

We have decided we should be charging for the actual value Lowdefy provides. This incentivises us to create a better platform, and not waste limited resources building services that another company already does well.

When selecting a pricing model for Lowdefy, we considered both user and usage based models. A flat fee per user has the advantage of simplicity and cost predictability, however, for us, it would exclude important use cases, for example - building apps with a large infrequent user base. Alternatively, a pure usage base model, for example - charging per request, would be too abstract for customers and cost predictability would be difficult. Ideally, we’d want to use a pricing model that implements the best aspects of both these models.

[Appsmith uses a pricing model](https://www.appsmith.com/pricing) that we believe does this really well. They charge for each hour that a user uses an app, capped at a maximum amount per user per month. We like this approach because:

- Predicting the maximum cost is easy as with popular user-based pricing models.
- Infrequent app users pay very little, thus allowing for the large infrequent user base use-case.
- This model scales to zero with no usage.
- It is easy to comprehend.

Ultimately we want our pricing model to align with the value Lowdefy creates, while being open to most use cases to promote Lowdefy adoption.

## New  Lowdefy Pricing Tiers

We’ll be implementing the following pricing tiers:

### Free Tier

- No restrictions will be placed on the development server.
- You will only be able to use user authentication and authorisation in production if it is for non-commercial use. We consider internal tools created and used by a company as commercial use.
- Apps in the free tier may include a “Made with Lowdefy” advertisement.
- Public pages will remain free.

### Paid Tier

- You can use user authentication and authorisation for commercial use.
- You will be charged per hour per user, capped to a maximum amount.
- All usage of public pages will remain free.

### Enterprise Tier

- An offline license key will be provided, allowing for air-gapped deployments.
- This tier will have custom pricing.

## Open Source  and Licensing

~We value the transparency provided by Open Source software and the trust that it builds. Therefore, we would prefer that Lowdefy remains Open Source. However, to implement our pricing plan, we may need to move to a source-available license.~

~We are considering licensing Lowdefy under the MariaDB [Business Source License](https://mariadb.com/bsl11/). This license is used by [Hashicorp](https://www.hashicorp.com/bsl), [Sentry](https://open.sentry.io/licensing/), and [CockroachDB](https://www.cockroachlabs.com/docs/stable/licensing-faqs). It allows us to license the project under one well known and understood license, instead of complicated dual licensing. The Business Source License is not certified as an open-source license but most of the Open Source Initiative (OSI) criteria are met.~

 We will release Lowdefy under a dual license. The majority of the project will remain licensed under the Apache License 2.0. The paid components of the project will be released under the MariaDB Business Source License 1.1.

## Next Steps

We consider the Lowdefy Version 4 release candidate versions production ready. We are currently working on implementing pricing to be included in the official V4 release. We would like to engage with the community on this, so please give us your feedback and thoughts on this discussion or reach out to hello@lowdefy.com.

---
Edit: Removed incorrect section on Open Source definition.
Edit: Update to license plans.

**Comment by Joffcom** [comment]
Sounds like a good plan to me, I look forward to finding out what the pricing actually is :)

**Comment by temtechie** [comment]
That is a good approach, But sacrosanct to keep the price fair enough within the range of competitors pricing in the market.
That way , you can give users more opportunities to choose over your product than that of other competitors. 

Possibly gather some prices and determine what is fair enough . just my opinion though. Thanks

**Comment by rodolfototaro** [comment]
Hi @SamTolmay,
Did you already evaluated the AGPL license? It is already used i.e. by Grafana or Minio projects. AGPL allows users to continue to use your opensource version of the software and contribute to it eventually with a contribution agreement that gives you rights for using contributions within your commercial version of the product. At the same time AGPL protects you against competitors that want to use the software for a proprietary SaaS without contributions to the code.
Last but not least, pay attention to the forks: see OpenTOFU opensource fork of Terraform by Hashicorp that now is a the Linux Foundation Project

**Comment by IvanDrag0** [comment]
Hi @SamTolmay, 

I'm glad that Lowdefy is starting to adopt a pricing model. This would definitely help the project grow and not rely on just the community for funding and support. Since I'm currently evaluating Lowdefy for some internal projects for my company, I do have a few questions regarding the licensing tiers:

- The Paid Tier mentions that you will charge per hour per use. Is that only for the cloud version or also for the self hosted one?
- A big use case for me is offline/airgapped use. Would that something that will only be supported in the enterprise version or also the other versions/tiers?
- Have you defined what what are the business/use requirements differences between the community, paid, and enterprise tiers (at what point would one have to move from one to the other)?
- Are you working on publishing a price matrix with the difference in features between the tiers?

**Reply by SamTolmay** [reply]
Thanks @IvanDrag0 

We are excited for the growth that a paid model can bring, we have a lot of ideas to improve the platform.

We won't be offering a cloud hosted service right now, so the pricing will be for self-hosted apps. We would need to create a service to host Lowdefy apps that competes in a very competitive space with services like Vercel, Fly.io, Netlify, Digital Ocean, Cloudflare, Amazon Web Services, Google Cloud Platform and Microsoft Azure. To do this would require a substantial development effort to create a service with a worse user experience and less features. This would also incentivise us to create a product that does not deploy easily using these services. Therefore, we decided it would be better to ensure that Lowdefy can easily be deployed or self-hosted, and to let hosting providers do what they do well.

Offline/airgapped use will be part of the enterprise tier. To be able to charge based on usage we need to track anonymised usage, and validate license keys with a server as they will expire if invoices are not paid. Enterprise tier users will receive a signed license key that can be validated offline.

We have a work-in-progress pricing page for the website which you can view a preview deployment of here:
https://lowdefy-website-git-website-pricing-page-lowdefy.vercel.app/pricing
This is not final, we have not yet decided on pricing, but should give a good idea of where we are headed.

**Comment by raffaeleiavazzo85** [comment]
Hello, I'm a teacher and computer engineer. I wanted to know if the open-source version of Lowdefy can also be used for commercial purposes by downloading the code from GitHub. In other words, I would like to use Lowdefy to develop web apps and profit from them.
